src/power_grid_model/_core/power_grid_meta.py

```diff
@@ -205,3 +205,40 @@ def initialize_array(
         dtype=power_grid_meta_data[data_type][component_type].dtype,
         order="C",
     )
+
+
+def attribute_dtype(
+    data_type: DatasetTypeLike, component_type: ComponentTypeLike, attribute: AttributeType
```

Comment:
I am unsure, but shouldn't we use `DatasetTypeVar` and `ComponentTypeLike` instead? Also, we expose these two in the docs, whereas the initial two we don't.

Reply:
We have: 
```
DatasetTypeLike: TypeAlias = DatasetType | str
DatasetTypeVar = TypeVar("DatasetTypeVar", bound=DatasetTypeLike)  # helper used for type deduction

ComponentTypeLike: TypeAlias = ComponentType | str
ComponentTypeVar = TypeVar("ComponentTypeVar", bound=ComponentTypeLike)  # helper used for type deduction
```

I basically used the same as initialize_array.
Dont remember the exact reasoning back then.